src/fidesops/task/filter_element_match.py

```diff
@@ -0,0 +1,127 @@
+import copy
+import logging
+from collections import defaultdict
+
+from typing import List, Any, Dict
+
+import pydash
+
+from fidesops.graph.config import FieldPath
+from fidesops.task.refine_target_path import (
+    build_incoming_refined_target_paths,
+    DetailedPath,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def filter_element_match(
+    row: Dict[str, Any], query_paths: Dict[FieldPath, List[Any]]
+) -> Dict[str, Any]:
+    """
+    Modifies row in place to only contain matched elements or matched embedded documents within arrays.
+
+    :param row: Record retrieved from a dataset
+    :param query_paths: FieldPaths mapped to query values
+    :return: A modified record with array elements potentially eliminated if array data was targeted by a query path
+
+    :Example:
+    The given row was retrieved from a dataset because an element in row["A"] matched 2 or an element in
+    row["C"]["D"] matched 5.  row["A"] is filtered to just contain the matching element, and row["C"] is filtered
+    to just contain the objects where "D" = 5.
+
+    filter_element_match(
+        row={"A": [1, 2, 3], "B": 2, "C": [{"D": 3, "E": 4}, {"D": 5, "E": 6}, {"D": 5, "E": 7}]},
+        query_paths={FieldPath("A"): [2], FieldPath("C, "D"): 5}
+    )
+
+    {"A": [2], "B": 2, "C": [{"D": 5, "E": 6}, {"D": 5, "E": 7}]}
```

Comment:
You've done a great job explaining what's being filtered here, and how that works. It might also be worth stating that paths not in `query_paths` won't be filtered at all? I was initially confused as to why `"B": 2` was in the results for instance

Reply:
An important point too is that it's only array elements that are being removed, so if {"B": 2} was in the `query_path`, there should be no effect. The goal of `filter_element_match` is to remove non-matched elements in _arrays_.

Reply:
I added a few points that this is for arrays, this docstring is being rewritten too in the array erasure PR as some of these methods are being used for multiple purposes.